### PR TITLE
Add timestamp and biddercode to event URL - refactored

### PR DESCRIPTION
**Pull request by rpanchyk**
See parent PR https://github.com/rubicon-project/prebid-server-java/pull/635

### CODE REVIEW

File: src/main/java/org/prebid/server/events/EventUtil.java [hunk 1]
```diff
@@ -88,10 +81,14 @@ public static void validateAnalytics(RoutingContext context) {
...
 
     public static void validateTimestamp(RoutingContext context) {
-        final String timestamp = context.request().params().get(TIMESTAMP_PARAMETER);
-        if (StringUtils.isBlank(timestamp)) {
-            throw new IllegalArgumentException(String.format(
-                    "Timestamp '%s' is required query parameter and can't be empty", TIMESTAMP_PARAMETER));
+        final String timestamp = StringUtils.stripToNull(context.request().params().get(TIMESTAMP_PARAMETER));
+        if (timestamp != null) {
+            try {
+                Long.parseLong(timestamp);
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/events/EventUtil.java:87]: `NumberUtils.isDigits()` or `StringUtils.isDigits()`, (but we still need to check for null bc this value is not required)

File: src/main/java/org/prebid/server/cache/CacheService.java [hunk 2]
```diff
@@ -168,32 +168,36 @@ public String getCachedAssetURLTemplate() {
...
-                final PutObject updatedPutObject = putObject.toBuilder().value(new TextNode(updatedVastValue)).build();
-                updatedPutObjects.add(updatedPutObject);
-            } else {
-                updatedPutObjects.add(putObject);
+                final String updatedVastXml = modifyVastXml(value.asText(), putObject.getBidid(),
+                        putObject.getBidder(), accountId, putObject.getTimestamp());
+                builder.value(new TextNode(updatedVastXml)).build();
             }
+
+            updatedPutObjects.add(builder.build());
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/cache/CacheService.java:200]: suggest
```
   for (PutObject putObject : putObjects) {

            JsonNode value = putObject.getValue();
            if (biddersAllowingVastUpdate.contains(putObject.getBidder()) && value != null) {
                final String updatedVastXml = modifyVastXml(value.asText(), putObject.getBidid(),
                        putObject.getBidder(), accountId, putObject.getTimestamp());
                value = new TextNode(updatedVastXml);
            }

            updatedPutObjects.add(putObject.toBuilder()
                    // remove "/vtrack" specific fields
                    .bidid(null)
                    .bidder(null)
                    .timestamp(null)
                    
                    .value(value)
                    .build());
        }
```

File: src/test/java/org/prebid/server/cache/CacheServiceTest.java [hunk 3]
```diff
@@ -1072,35 +1071,63 @@ public void cachePutObjectsShouldModifyVastAndCachePutObjects() throws IOExcepti
         // given
         final PutObject firstPutObject = PutObject.builder()
                 .type("xml")
-                .bidid("biddid1")
+                .bidid("bidId1")
```
Comment (DGarbar) [inline src/test/java/org/prebid/server/cache/CacheServiceTest.java:1074]: I'd also added `.timestamp(1231221L)` to test

**Comment by DGarbar** [comment]
And Please refactor ExchangeService method placement according to workflow

File: src/main/java/org/prebid/server/events/EventUtil.java (hunk 1 above)
Comment (rpanchyk) [reply src/main/java/org/prebid/server/events/EventUtil.java:87]: Both util methods doesn't guaranteed Long range value checking, so `NumberFormatException` can be thrown. Don't think they are useful here.

File: src/main/java/org/prebid/server/cache/CacheService.java (hunk 2 above)
Comment (rpanchyk) [reply src/main/java/org/prebid/server/cache/CacheService.java:200]: kind of taste-)

**Pull request merged**